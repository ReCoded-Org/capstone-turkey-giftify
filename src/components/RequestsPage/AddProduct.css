.addProductMainDiv {
  display: flex;
  flex-direction: column;
  flex-wrap: nowrap;
  justify-content: center;
  align-items: center;
  margin: 10% 0%;
}

.addProductMainDiv > form {
  display: flex;
  flex-direction: column;
  flex-wrap: nowrap;
  justify-content: space-around;
  align-items: center;
  align-content: stretch;
  background-color: #90e0ef;
  width: 65%;
  border-radius: 15px;
}

.addProductMainDiv > form > label {
  display: flex;
  flex-direction: column;
  flex-wrap: nowrap;
  justify-content: space-between;
  align-items: center;
  align-content: stretch;
  width: 65%;
  font-weight: bold;
  font-size: 20px;
}
.addProductMainDiv > form > label > select,
.addProductMainDiv > form > label > input,
.addProductMainDiv > form > label > textarea {
  width: 100%;
}

.addProductMainDiv > form > label > span {
  margin: 3% 0%;
}

.buttonDivAddProduct {
  width: 100%;
  display: flex;
  flex-direction: row;
  flex-wrap: nowrap;
  justify-content: center;
  align-items: center;
  align-content: stretch;
  margin: 5% 0%;
}

@media only screen and (min-width: 750px) and (max-width: 1009px) {
  .addProductMainDiv > form > label {
    width: 50%;
    font-weight: bold;
    font-size: 17.5px;
  }

  .addProductMainDiv > form {
    width: 60%;
  }
}

@media only screen and (min-width: 120px) and (max-width: 749px) {
  .addProductMainDiv > form > label {
    width: 80%;
    font-weight: bold;
    font-size: 17.5px;
  }

  .addProductMainDiv > form {
    width: 85%;
  }
}
