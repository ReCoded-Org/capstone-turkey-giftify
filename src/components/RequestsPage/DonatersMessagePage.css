.donatersMessageContainer {
  display: flex;
  flex-direction: row;
  flex-wrap: nowrap;
  justify-content: space-evenly;
  align-items: stretch;
  align-content: stretch;
  width: 100%;
}

.mainProductDetails {
  background: #90e0ef;
  border-radius: 15px;
  width: 45%;
}

.productDetails {
  display: flex;
  flex-direction: column;
  flex-wrap: nowrap;
  justify-content: space-around;
  align-items: baseline;
  align-content: stretch;
  margin: 5% 0%;
}

.productDetailsText {
  margin: 0% 0% 0% 7%;
}

.productDetailsText > span {
  margin: 0% 0% 1% 2%;
  font-weight: bold;
}

.productDetailsText > p {
  background: #f8f9fa;
  border: 1px solid #006eff;
  box-sizing: border-box;
  border-radius: 15px;
  padding: 3%;
  width: 90%;
}

.productDetailsImage {
  display: flex;
  flex-direction: column;
  flex-wrap: nowrap;
  justify-content: space-around;
  align-items: center;
  align-content: stretch;
  width: 100%;
}

.productDetailsImage > img {
  width: 75%;
  border-radius: 15px;
  margin: 0% 0% 5% 0%;
}

.messageArea {
  background: #90e0ef;
  border-radius: 15px;
  width: 45%;
}

.donatersMessageText {
  display: flex;
  flex-direction: column;
  flex-wrap: nowrap;
  justify-content: space-around;
  align-items: baseline;
  align-content: stretch;
  width: 100%;
  margin: 5%;
}

.donatersMessageText > span {
  margin: 0% 0% 1% 2%;
  font-weight: bold;
}

.donatersMessageText > p {
  background: #f8f9fa;
  border: 1px solid #006eff;
  box-sizing: border-box;
  border-radius: 15px;
  padding: 3%;
  width: 90%;
}

@media only screen and (min-width: 120px) and (max-width: 531px) {
  .donatersMessageContainer {
    display: flex;
    flex-direction: column;
    flex-wrap: nowrap;
    justify-content: space-around;
    align-items: center;
    align-content: stretch;
  }

  .mainProductDetails {
    width: 85%;
  }

  .messageArea {
    margin-top: 5%;
    width: 85%;
  }
}
