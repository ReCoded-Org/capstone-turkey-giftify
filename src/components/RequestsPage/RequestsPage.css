.requstsPageContainer{
	display: flex;
	flex-direction: row;
	flex-wrap: wrap;
	justify-content: flex-start;
	align-items: stretch;
	align-content: space-evenly;
}

.cardInRequestPage{
    width: 30%;
    margin: 1%;
    display: flex;
	flex-direction: column;
	flex-wrap: nowrap;
	justify-content: space-between;
	align-items: center;
	align-content: stretch;
}

.CardUpperPart{
    background-color: #ADE8F4;
    width: 100%;
}

.card-title{
    font-family: Roboto;
    font-style: normal;
    font-weight: 500;
    font-size: 24px;
    line-height: 34px;
    text-align: center;
}

.card-img-top{
    max-width: 350px;
    height: 70%;
}

/* FIX CARD IMAGE HEIGHT */

@media only screen and (min-width: 532px) and (max-width: 991px){
    .cardInRequestPage{
        width: 45%;
        margin: 1%;
    }
}

@media only screen and (min-width: 120px) and (max-width: 531px){
    .requstsPageContainer{
        display: flex;
        flex-direction: row;
        flex-wrap: wrap;
        justify-content: center;
        align-items: center;
        align-content: space-around;
    }

    .cardInRequestPage{
        width: 85%;
        margin: 1%;
    }
}