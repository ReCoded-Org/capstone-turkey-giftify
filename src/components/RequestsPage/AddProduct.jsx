import React, { useState, useEffect } from "react";
import db from "./../../firebase";
import "./AddProduct.css";
import { Button } from "react-bootstrap";
import { useHistory } from "react-router-dom";

const AddProduct = () => {
  // eslint-disable-next-line
  const [userID, setUserID] = useState("2");
  const [itemName, setItemName] = useState("");
  const [productCategory, setProductCategory] = useState({ value: "books" });
  const [description, setDescription] = useState("");
  const [condition, setCondition] = useState("");
  const [image, setImage] = useState("");
  const [categoriesName, setCategoriesName] = useState([]);
  const history = useHistory();

  const getCategories = async () => {
    const collection = await db.collection("categories").get();
    const arrayOfDocs = collection.docs.map((doc) => {
      return { id: doc.id, ...doc.data() };
    });
    setCategoriesName(arrayOfDocs);
  };

  const handleSubmit = (e) => {
    e.preventDefault();
    db.collection("categories")
      .doc(productCategory.value)
      .collection("items")
      .add({
        userID,
        itemName,
        description,
        condition,
        image,
      })
      .then(history.push("/requestsPage"));
  };

  const handleChange = (e) => {
    setProductCategory({ value: e.target.value });
  };

  useEffect(() => {
    getCategories();
    // eslint-disable-next-line
  }, []);

  return (
    <div className="addProductMainDiv">
      <form onSubmit={handleSubmit}>
        <label>
          <span>Product's Image:</span>
          <input
            type="text"
            onChange={(e) => setImage(e.target.value)}
            value={image}
          />
        </label>
        <label>
          <span>Product's Name:</span>
          <input
            type="text"
            onChange={(e) => setItemName(e.target.value)}
            value={itemName}
          />
        </label>
        <label>
          <span>Product's Category:</span>
          <select value={productCategory.value} onChange={handleChange}>
            {categoriesName.map((info) => {
              return (
                <option key={info.id} value={info.id}>
                  {info.id.charAt(0).toUpperCase() + info.id.slice(1)}
                </option>
              );
            })}
          </select>
        </label>
        <label>
          <span>Product's Short Brief:</span>
          <textarea
            type="text"
            onChange={(e) => setDescription(e.target.value)}
            style={{ resize: "vertical" }}
            value={description}
          />
        </label>
        <label>
          <span>Product's Condition:</span>
          <input
            type="text"
            onChange={(e) => setCondition(e.target.value)}
            value={condition}
          />
        </label>
        <div className="buttonDivAddProduct">
          <Button variant="primary" type="submit">
            Add Product
          </Button>
        </div>
      </form>
    </div>
  );
};

export default AddProduct;
