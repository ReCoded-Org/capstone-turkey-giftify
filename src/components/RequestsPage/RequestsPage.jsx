import React, { useState, useEffect } from "react";
import db from "./../../firebase";
import { Card } from "react-bootstrap";
import { ArrowRight } from "react-bootstrap-icons";
import { Link } from "react-router-dom";
import "./RequestsPage.css"

const RequestsPage = () => {
  // need redux
  // hangi user loggedin
  // eslint-disable-next-line
  const [userId, setUserId] = useState("2");
  const [usersProducts, setUsersProducts] = useState([]);
  let categoriesName = [];

  const getElements = async () => {
    const categoriesPath = db.collection("categories");
    await categoriesPath.get().then((querySnapshot) => {
      querySnapshot.forEach((doc) => {
        categoriesName.push(doc.id);
        categoriesName.map((category) => {
          return categoriesPath
            .doc(`${category}`)
            .collection("items")
            .where("userID", "==", userId)
            .get()
            .then((querySnapshot) => {
              querySnapshot.forEach((doc) => {
                setUsersProducts((prevState) => {
                  return {
                    ...prevState,
                    [doc.id]: [doc.data().itemName, doc.data().image, doc.id],
                  };
                });
              });
            });
        });
      });
    });
  };

  useEffect(() => {
    getElements();
    // eslint-disable-next-line
  }, []);

  console.log("usersProducts: ", usersProducts);

  return (
    <div className="requstsPageContainer">
      {Object.values(usersProducts).map((info) => (
        <Card key={info[0]} className="cardInRequestPage">
          <Card.Img variant="top" src={info[1]} />
          <Card.Body className="CardUpperPart">
            <Card.Title>{info[0]}</Card.Title>
            <Card.Link>
              <Link to={`/donatersMessagePage/${info[2]}`}>See Details <ArrowRight /></Link>
            </Card.Link>
          </Card.Body>
        </Card>
      ))}
    </div>
  );
};

export default RequestsPage;
