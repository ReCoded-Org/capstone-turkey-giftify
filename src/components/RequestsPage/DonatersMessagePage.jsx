import React, { useState, useEffect } from "react";
import { Card } from "react-bootstrap";
import "./DonatersMessagePage.css";
import { useParams } from "react-router";
import db from "./../../firebase";

const DonatersMessagePage = () => {
  const { id } = useParams();

  const [usersMessages, setUsersMessages] = useState([]);
  const [productDetails, setProductDetails] = useState([]);
  let categoriesName = [];

  const getRequestsMessage = async () => {
    const categoriesPath = db.collection("categories");
    await categoriesPath.get().then((querySnapshot) => {
      querySnapshot.forEach((doc) => {
        categoriesName.push(doc.id);
        categoriesName.map((category) => {
          return categoriesPath
            .doc(`${category}`)
            .collection("items")
            .doc(`${id}`)
            .collection("requests")
            .get()
            .then((querySnapshot) => {
              querySnapshot.forEach((doc) => {
                console.log("docc: ", doc.data());
                setUsersMessages((prevState) => {
                  return {
                    ...prevState,
                    [doc.id]: [doc.data().reason, doc.data().userId],
                  };
                });
              });
            });
        });
      });
    });
  };

  const getProductDetails = async () => {
    const categoriesPath = db.collection("categories");
      categoriesName.map((category) => {
        return categoriesPath
        // use category
          .doc(`${category}`)
          .collection("items")
          .doc(`${id}`)
          .get()
          .then((querySnapshot) => {
            console.log("querySnapshot.data(): ", querySnapshot.data());
            setProductDetails(querySnapshot.data());
          });
      });
  };

  console.log("usersMessages: ", usersMessages);
  console.log("productDetails: ", productDetails);

  useEffect(() => {
    getRequestsMessage();
    getProductDetails();
    // eslint-disable-next-line
  }, []);

  return (
    <div className="donatersMessageContainer">
      <div className="mainProductDetails">
        <p>SADECE CIGLIK ATIYORUM ARTIK</p>
        {/* returns undefined state and after an action call returns the state correctly */}
        {/* {Object.values(productDetails).map((info) => (
                    <Card key={info}>

                    </Card>
                ))} */}
      </div>
      <div className="messageArea">
        {Object.values(usersMessages).map((info) => (
          <div className="donatersMessageText">
            <span>User ID: {info[1]}</span>
            <p>{info[0]}</p>
          </div>
        ))}
      </div>
    </div>
  );
};

export default DonatersMessagePage;
