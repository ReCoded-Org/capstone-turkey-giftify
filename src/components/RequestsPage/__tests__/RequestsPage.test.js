import React from "react";
import RequestsPage from "./../RequestsPage";
import { cleanup } from "@testing-library/react";
import renderer from "react-test-renderer";

afterEach(cleanup);

it("RequestsPage rendered correctly", () => {
  const tree = renderer.create(<RequestsPage />).toJSON();
  expect(tree).toMatchSnapshot();
});
