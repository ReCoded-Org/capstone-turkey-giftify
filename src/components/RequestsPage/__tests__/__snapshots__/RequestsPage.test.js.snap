// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RequestsPage rendered correctly 1`] = `
<div
  className="requstsPageContainer"
/>
`;
